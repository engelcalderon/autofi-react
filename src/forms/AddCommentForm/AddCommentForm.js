import React from 'react';
import { string } from 'prop-types';
import classNames from 'classnames';
import { Form, Field } from 'react-final-form'

import css from './AddCommentForm.module.css';

const AddCommentForm = props => {

    return (
        <Form
            {...props}
            render={renderProps => {

                const {
                    className,
                    handleSubmit,
                    form
                } = renderProps;

                const classes = classNames(css.root, className);

                return (
                    <form onSubmit={values => handleSubmit(values, form)} className={classes}>
                        <h2>Add a comment</h2>
                        <div className={css.container}>
                            <Field
                                className={css.field}
                                name="name"
                                component="input"
                                placeholder="Name"
                            />
                            <Field
                                className={css.field}
                                name="email"
                                component="input"
                                placeholder="Email"
                                type="email"
                            />
                            <Field
                                className={css.field}
                                name="body"
                                component="textarea"
                                placeholder="Type a comment..."
                                rows={2}
                            />
                        </div>
                        <button className={css.submit}>Send</button>
                    </form>
                );
            }}
        />
    )
};

AddCommentForm.defaultProps = {
    className: null
}

AddCommentForm.propTypes = {
    className: string
}

export default AddCommentForm;

