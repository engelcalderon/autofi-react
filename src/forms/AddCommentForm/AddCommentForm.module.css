
.container {
    display: flex;
    flex-wrap: wrap;
}

.field {
    flex-basis: calc(50% - 27px);
    margin-right: 12px;
    margin-bottom: 15px;
    height: 25px;
    border-radius: 5px;
    border: 1px solid gray;
    padding: 5px;
}

.submit {
    width: 200px;
    height: 50px;
    border: none;
    font-weight: bold;
    border-radius: 5px;
    cursor: pointer;
    outline: none;
}