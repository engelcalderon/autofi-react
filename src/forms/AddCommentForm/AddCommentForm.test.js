import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddCommentForm from './AddCommentForm';

describe('AddCommentForm', () => {
    it('matches snapshot', () => {
        const props = {
            className: null,
            onSubmit: () => null,
        }
        const tree = toJson(mount(<AddCommentForm {...props} />), { mode: 'deep' });
        expect(tree).toMatchSnapshot();
    })
});