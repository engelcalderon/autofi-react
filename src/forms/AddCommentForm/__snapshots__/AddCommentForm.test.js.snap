// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentForm matches snapshot 1`] = `
<form
  className="root"
  onSubmit={[Function]}
>
  <h2>
    Add a comment
  </h2>
  <div
    className="container"
  >
    <input
      className="field"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Name"
      value=""
    />
    <input
      className="field"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
    <textarea
      className="field"
      name="body"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Type a comment..."
      rows={2}
      value=""
    />
  </div>
  <button
    className="submit"
  >
    Send
  </button>
</form>
`;
